src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -3644,16 +3644,41 @@ void addToStringMethod(MethodSymbol printMethod)
 
                 if (!memberSignatures.TryGetValue(targetMethod, out Symbol? existingToStringMethod))
                 {
-                    var toStringMethod = new SynthesizedRecordToString(this, printMethod, memberOffset: members.Count, diagnostics);
-                    members.Add(toStringMethod);
+                    var baseToStringMethod = getBaseToStringMethod();
+                    if (baseToStringMethod is null || !baseToStringMethod.IsSealed)
+                    {
+                        var toStringMethod = new SynthesizedRecordToString(this, printMethod, memberOffset: members.Count, diagnostics);
+                        members.Add(toStringMethod);
+                    }
                 }
                 else
                 {
                     var toStringMethod = (MethodSymbol)existingToStringMethod;
                     if (!SynthesizedRecordObjectMethod.VerifyOverridesMethodFromObject(toStringMethod, SpecialType.System_String, diagnostics) && toStringMethod.IsSealed && !IsSealed)
                     {
-                        diagnostics.Add(ErrorCode.ERR_SealedAPIInRecord, toStringMethod.Locations[0], toStringMethod);
+                        MessageID.IDS_FeatureSealedRecordToString.CheckFeatureAvailability(
+                            diagnostics,
+                            this.DeclaringCompilation,
+                            toStringMethod.Locations[0]);
+                    }
+                }
+
+                MethodSymbol? getBaseToStringMethod()
+                {
+                    var tmp = this.BaseTypeNoUseSiteDiagnostics;
+                    while (tmp is not null)
+                    {
+                        var method = tmp.GetSimpleNonTypeMembers(WellKnownMemberNames.ObjectToString)
+                            .OfType<MethodSymbol>()
```

Comment:
>OfType [](start = 29, length = 6)

I think a regular loop through members would be preferred other the Linq query in this case.

Reply:
Done